.github/workflows/publish.yml

```
@@ -27,6 +27,9 @@ jobs:
     needs: release-please
     if: ${{ needs.release-please.outputs.release_created }}
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
```

I don't think we need write
```suggestion
      contents: read
```
Any chance you could move the other permission set off of the top-level?